uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/models/HabitListTest.kt

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (C) 2016-2021 Álinson Santos Xavier <git@axavier.org>
+ *
+ * This file is part of Loop Habit Tracker.
+ *
+ * Loop Habit Tracker is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * Loop Habit Tracker is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isoron.uhabits.core.models
+
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertFalse
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertNull
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert.assertThat
+import org.isoron.uhabits.core.BaseUnitTest
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import java.io.IOException
+import java.io.StringWriter
+import java.util.ArrayList
+
+class HabitListTest : BaseUnitTest() {
+    @get:Rule
+    var thrown = ExpectedException.none()!!
+    private lateinit var habitsArray: ArrayList<Habit>
+    private lateinit var activeHabits: HabitList
+    private lateinit var reminderHabits: HabitList
+
+    @Throws(Exception::class)
+    override fun setUp() {
+        super.setUp()
+        habitsArray = ArrayList()
+        for (i in 0..9) {
+            val habit = fixtures.createEmptyHabit()
+            habitList.add(habit)
+            habitsArray.add(habit)
+            if (i % 3 == 0) habit.reminder = Reminder(8, 30, WeekdayList.EVERY_DAY)
+        }
+        habitsArray[0].isArchived = true
+        habitsArray[1].isArchived = true
+        habitsArray[4].isArchived = true
+        habitsArray[7].isArchived = true
+        activeHabits = habitList.getFiltered(HabitMatcherBuilder().build())
+        reminderHabits = habitList.getFiltered(
+            HabitMatcherBuilder()
+                .setArchivedAllowed(true)
+                .setReminderRequired(true)
+                .build()
+        )
+    }
+
+    @Test
+    fun testSize() {
+        assertThat(habitList.size(), CoreMatchers.equalTo(10))
+        assertThat(activeHabits.size(), CoreMatchers.equalTo(6))
+        assertThat(reminderHabits.size(), CoreMatchers.equalTo(4))
+    }
+
+    @Test
+    fun testGetByPosition() {
+        assertThat(
+            habitList.getByPosition(0),
+            CoreMatchers.equalTo(
+                habitsArray[0]
+            )
+        )
```

Comment:
Please put these in a single line. I think ktlint breaks lines, but doesn't merge them back.